driver-async/src/main/com/mongodb/async/client/internal/KeyManagementService.java

```diff
@@ -61,7 +56,10 @@ void decryptKey(final MongoKeyDecryptor keyDecryptor, final SingleResultCallback
     }
 
     private void streamOpen(final MongoKeyDecryptor keyDecryptor, final SingleResultCallback<Void> callback) {
-        final Stream stream = streamFactory.create(new ServerAddress(keyDecryptor.getHostName(), port));
+        ServerAddress serverAddress = keyDecryptor.getHostName().contains(":")
```

Comment:
The host name can contain a ":<port>" and that case was not handled before.  The `getServerAddress` method below would fail because the `ServerAddress` constructor doesn't allow a host name with a port when a port is also specified.

Reply:
No, not a bug.  The port field is just the default value.  I can change its name to defaultPort to make it clear.

Reply:
Done: changed property name